app/src/main/java/com/babylon/wallet/android/presentation/dialogs/info/InfoDialog.kt

```diff
@@ -0,0 +1,261 @@
+            val destination = child.findChildOfType(LINK_DESTINATION)
+                ?.getTextInNode(content)
+                ?.toString()
+            val linkLabel = child.findChildOfType(LINK_LABEL)
+                ?.getTextInNode(content)?.toString()
+            val annotation = destination ?: linkLabel
+            if (annotation != null) pushStringAnnotation(MARKDOWN_TAG_URL, annotation)
+
+            pushStyle(
+                SpanStyle(
+                    fontWeight = FontWeight.Bold,
+                    color = Color.Blue
+                )
+            )
+
+            append(linkText?.getTextInNode(content).toString())
+            pop()
+            true // return true to consume this ASTNode child
+        } else {
+            false
+        }
+    }
+
+    Column(
+        modifier = modifier
+            .background(RadixTheme.colors.defaultBackground)
+            .fillMaxHeight(0.5f)
+            .verticalScroll(scrollState)
+            .padding(horizontal = RadixTheme.dimensions.paddingLarge)
+            .padding(top = RadixTheme.dimensions.paddingSmall)
+            .padding(bottom = RadixTheme.dimensions.paddingDefault),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        drawableRes?.let {
+            Icon(
+                painterResource(id = drawableRes),
+                tint = Color.Unspecified,
+                contentDescription = null
+            )
+        }
+
+        CompositionLocalProvider(
+            LocalUriHandler provides object : UriHandler {
+                override fun openUri(uri: String) {
+                    if (uri.contains(GLOSSARY_ANCHOR)) {
+                        onGlossaryItemClick(uri.drop(GLOSSARY_ANCHOR.length))
+                    } else {
+                        context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(uri)))
+                    }
+                }
+            }
+        ) {
+            Markdown(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(),
+                typography = markdownTypography(
+                    h2 = RadixTheme.typography.title,
+                    paragraph = RadixTheme.typography.body1Regular,
+
+                ),
+                colors = markdownColor(
+                    linkText = RadixTheme.colors.blue2
+                ),
+                padding = markdownPadding(
+                    block = RadixTheme.dimensions.paddingSmall,
+                ),
+                components = markdownComponents(
+                    heading2 = customHeading2
+                ),
+                annotator = linkAnnotator,
+                content = markdownContent ?: stringResource(id = R.string.empty),
+            )
+        }
+    }
+}
+
+// Helper function to drop the first and last element
+// in order to not render the brackets of a link
+internal fun List<ASTNode>.innerList(): List<ASTNode> = this.subList(1, this.size - 1)
```

Comment:
I don't know if that might ever be the case, but checking the `from` and `to` indices against the list size is always a good idea to prevent `ArrayIndexOutOfBoundException`.

Reply:
this was taken from inside the library. 
I can use `take` in order to avoid the exception. What do you think?